app/v2/services/AuditingService.scala

```diff
@@ -33,15 +33,21 @@ import scala.util.control.NonFatal
 @ImplementedBy(classOf[AuditingServiceImpl])
 trait AuditingService {
 
-  def audit(auditType: AuditType, source: Source[ByteString, _], contentType: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Unit]
+  def audit(auditType: AuditType, source: Source[ByteString, _], contentType: String, contentLength: Long = 0)(implicit
```

Comment:
I know this has been merged and all... but what does a `contentLength` of 0 mean? This sends the header with a content length of 0... and what does the auditing service do with that?

I checked -- and it basically assumes that the message is small. Is that really what we should be doing? Yes, it does that in the absence of a header too, but at least there the intent is crystal clear. Sending a `X-ContentSize` header of `0` for a non-zero body isn't right.

This is something that really should be an `Option` and then not send the header if we can't supply the size. We're now relying on a magic number to determine behaviour, rather than the existence or absence of a header. 

This needs rectifying because this isn't a great thing to do.